src/javascript/components/SitemapPanelApp.jsx

```diff
@@ -0,0 +1,277 @@
+        // eslint-disable-next-line no-unused-vars
+        onSubmit: values => {
+            if (!sitemapMixinEnabled) {
+                gqlUtilities.gqlMutate(client, gqlMutations.AddMixin, {
+                    pathOrId: `/sites/${dxContext.siteKey}`,
+                    mixins: ['jseomix:sitemap']
+                });
+                setSitemapMixinEnabled(prevState => !prevState);
+                setSnackbarIsOpen(true);
+            }
+
+            gqlUtilities.gqlMutate(client, gqlMutations.mutateProperty, {
+                pathOrId: `/sites/${dxContext.siteKey}`,
+                propertyName: 'sitemapIndexURL',
+                propertyValue: formik.values.sitemapIndexURL
+            });
+            gqlUtilities.gqlMutate(client, gqlMutations.mutateProperty, {
+                pathOrId: `/sites/${dxContext.siteKey}`,
+                propertyName: 'sitemapCacheDuration',
+                propertyValue: formik.values.sitemapCacheDuration
+            });
+        }
+    });
+
+    useEffect(() => {
+        gqlUtilities.gqlQuery(client, gqlQueries.GetNodeMixin, {pathOrId: `/sites/${dxContext.siteKey}`,
+            mixinsFilter: {
+                filters: [
+                    {fieldName: 'name', value: 'jseomix:sitemap'}
+                ]
+            }
+        }).then(data => {
+            if (data?.data?.jcr?.nodeByPath?.mixinTypes?.length > 0) {
+                setSitemapMixinEnabled(true);
+            }
+        });
+
+        gqlUtilities.gqlQuery(client, gqlQueries.GetProperties, {pathOrId: `/sites/${dxContext.siteKey}`,
+            propertyNames: ['sitemapIndexURL', 'sitemapCacheDuration']
+        }).then(data => {
+            const properties = data?.data?.jcr?.nodeByPath?.properties;
+            if (properties.length > 0) {
+                properties.forEach(property => {
+                    if (property.name === 'sitemapIndexURL') {
+                        setSitemapIndexURL(property.value);
+                    } else if (property.name === 'sitemapCacheDuration') {
+                        setSitemapCacheDuration(property.value);
+                    }
+                });
+            }
+        });
+    }, [client, dxContext.siteKey]);
+
+    const onAcademyButtonClick = () => {
+        window.open('https://academy.jahia.com/documentation/enduser/jahia/8/advanced-authoring/seo/sitemap', '_blank');
+    };
+
+    const onOpenSitemapXMLButtonClick = url => {
+        window.open(url, '_blank');
+    };
+
+    const handleDialogOpen = (title, text, submitText) => {
+        setDialogInfo({
+            title: title,
+            text: text,
+            submitText: submitText
+        });
+        setDialogIsOpen(true);
+    };
+
+    const handleDialogClose = () => {
+        setDialogInfo(null);
+        setDialogIsOpen(false);
+    };
+
+    const handleSnackBarClose = () => {
+        setSnackbarIsOpen(false);
+    };
+
+    return (
+        <form onSubmit={formik.handleSubmit}>
```

Comment:
Would be nice if we could break jxs down into smaller more manageable pieces.

Reply:
Let me talk to you on what you have in mind (As right now it is already using base components)

Reply:
agree with this one. for starters, big chunks would be the header then the actual panel if possible. within the panel ui, the different subsections input can also be split out.

Reply:
@gflores-jahia 
If you look closely we are actually just calling one component **EX**: for header with override for different props.
Other than that if we are worried about those buttons are not too readable (debatable)
I am not too sure if it good to re-factor this as (those "components" are will not be re-usable if factored out)

As for the different panels there are some dependencies between each **EX**: as if URL is not specify then display for selection of sitemap URL will always be empty. (Introducing complexity for redux or use child component to manage state of separate branch node)

From my point of view we should only re-factor out if those component can be swap out individually functionally

Reply:
That is fine 
Updated for readability purpose